test/backup.ts

```diff
@@ -0,0 +1,515 @@
+      assert.strictEqual(result, undefined);
+    });
+
+    it('should rethrow other errors', async () => {
+      const err = {code: grpc.status.INTERNAL};
+      backup.getMetadata = async () => {
+        throw err;
+      };
+
+      try {
+        await backup.getExpireTime();
+        assert.fail('Should have rethrown error');
+      } catch (thrown) {
+        assert.deepStrictEqual(thrown, err);
+      }
+    });
+  });
+
+  describe('exists', () => {
+    it('should return true when backup info indicates backup exists', async () => {
+      const BACKUP_INFO_RESPONSE: GetMetadataResponse = [{}];
+      backup.getMetadata = async () => BACKUP_INFO_RESPONSE;
+
+      const result = await backup.exists();
+      assert.strictEqual(result, true);
+    });
+
+    it('should return false when backup does not exist', async () => {
+      backup.getMetadata = async () => {
+        throw {code: grpc.status.NOT_FOUND};
+      };
+
+      const result = await backup.exists();
+      assert.strictEqual(result, false);
+    });
+
+    it('should rethrow other errors', async () => {
+      const err = {code: grpc.status.INTERNAL};
+      backup.getMetadata = async () => {
+        throw err;
+      };
+
+      try {
+        await backup.exists();
+        assert.fail('Should have rethrown error');
+      } catch (thrown) {
+        assert.deepStrictEqual(thrown, err);
+      }
+    });
+  });
+
+  describe('updateExpireTime', () => {
+    const NEW_EXPIRE_TIME = new PreciseDate(1977, 1, 10);
+
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        backup: {
+          name: BACKUP_FORMATTED_NAME,
+          expireTime: NEW_EXPIRE_TIME.toStruct(),
+        },
+        updateMask: {
+          paths: ['expire_time'],
+        },
+      });
+
+      backup.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'updateBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await backup.updateExpireTime(NEW_EXPIRE_TIME);
+    });
+
+    it('should accept gaxOpts and a callback', async () => {
+      const options = {
+        timeout: 1000,
+      };
+
+      backup.request = config => {
+        assert.deepStrictEqual(config.gaxOpts, options);
+      };
+
+      await backup.updateExpireTime(NEW_EXPIRE_TIME, options, assert.ifError);
+    });
+
+    describe('error', () => {
+      const ERROR = new Error('Error.');
+      const API_RESPONSE = {};
+
+      beforeEach(() => {
+        backup.request = (config, callback: Function) => {
+          callback(ERROR, null, API_RESPONSE);
+        };
+      });
+
+      it('should execute callback with error & API response', done => {
```

Comment:
Same thought from earlier about losing "error" and "success" describes and tests complexity, and just use this single test.

Reply:
Agree but I chose to simplify it to the success case.